src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,167 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array = (T[]) new Object[DEFAULT_CAPACITY];
+    private int listLength;
+
     @Override
     public void add(T value) {
-
+        if (listLength < array.length) {
+            array[listLength] = value;
+            listLength++;
+        } else {
+            array = grow(array);
+            array[listLength] = value;
+            listLength++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + listLength);
+        }
+        if (index < array.length && index < listLength) {
```

Comment:
The condition 'index < array.length' is redundant because 'listLength' is never greater than 'array.length'. You can safely remove 'index < array.length' from the condition.